src/NuGet.Core/NuGet.PackageManagement/FileModifiers/XdtTransformer.cs

```diff
@@ -113,9 +113,10 @@ public class XdtTransformer : IPackageFileTransformer
                             // make sure we close the input stream immediately so that we can override 
                             // the file below when we save to it.
                             string path = FileSystemUtility.GetFullPath(msBuildNuGetProjectSystem.ProjectFullPath, targetPath);
-                            using (var reader = XmlReader.Create(path, GetXmlReaderSettings(LoadOptions.PreserveWhitespace)))
```

Comment:
Does this mean that we shouldn't have any https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlreader.create?view=net-5.0#System_Xml_XmlReader_Create_System_String_System_Xml_XmlReaderSettings_System_Xml_XmlParserContext_ calls?

Reply:
Yes, that is my understanding reading from the VS repo bugs (whose links I mentioned in the linked issue). Without this fix, `nuget.exe restore for packages.config` project fails with below error if the project is created under a folder whose name is `U1[]U2[]U3[]`

``` log
> NuGet.exe restore "C:\Users\xxx\source\repos\NuGet.Client\.test\work\26e269db\2faa8870\solution\U1[]U2[]U3[]"
MSBuild auto-detection: using msbuild version '17.0.0.46501' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
Could not find a part of the path 'C:\Users\xxx\source\repos\NuGet.Client\.test\work\26e269db\2faa8870\solution\U1[%EE%80%A5%EE%80%A6%EE%80%A7%EE%80%B8%EE%80%B9]U2[%EE%89%9A%EE%89%9B%EE%89%AC%EE%89%AD]U3[%EE%93%BE%EE%93%BF%EE%94%80%EE%94%8B%EE%94%8C]\a\packages.config'.

```